lib/internal/test_runner/runner.js

```diff
@@ -4,21 +4,18 @@ const {
   ArrayPrototypeFilter,
   ArrayPrototypeIncludes,
   ArrayPrototypeJoin,
-  ArrayPrototypePop,
   ArrayPrototypePush,
   ArrayPrototypeSlice,
   ArrayPrototypeSort,
   ObjectAssign,
   PromisePrototypeThen,
-  RegExpPrototypeSymbolSplit,
   SafePromiseAll,
   SafeSet,
-  StringPrototypeEndsWith,
 } = primordials;
 
-const { Buffer } = require('buffer');
 const { spawn } = require('child_process');
 const { readdirSync, statSync } = require('fs');
+const readline = require('readline/promises');
```

Comment:
`node:readline/promises` is not available on Node.js 16.x, given that we don't use any of the promisified APIs, we might as well use `node:readline`. Also, we should use destructuring in case the user overrides the value of `readline.createInterface`.
```suggestion
// TODO(aduh95): switch to internal/readline/interface when backporting to Node.js 16.x is no longer a concern.
const { createInterface } = require('readline');
```